### PR TITLE
Ability to disable/lock the android ViewPager scroll from props

### PR DESCRIPTION
**Pull request by kevinejohn**
This is a nice feature to have.

I've tested this by copying and renaming the ViewPager java and javascript files from the react-native repo and including them in a project. Whats the best way to test this directly from the repo?

### CODE REVIEW

**Comment by janicduplessis** [comment]
You can test it in the UIExplorer example. You should also add an example that uses the prop.

**Comment by kevinejohn** [comment]
I updated the code syntax to match the recently added Android ScrollView scrollEnabled:

https://github.com/facebook/react-native/commit/82b0df9d2a6163879647691235f01b3067c471b8

**Comment by astuetz** [comment]
@kevinejohn I personally wouldn't call the property ```scrollEnabled```, as usually when talking about a ```ViewPager``` I think most people use the word "swiping", not "scrolling". For our custom ```ViewPager```s, we simply use ```private boolean isLocked;```

**Comment by bestander** [comment]
Number of commits look wacky.
Maybe rebase?

**Comment by kevinejohn** [comment]
Yeah I rebased to master and force pushed. Is there a better way to do that so it doesn't look funny on github?

**Comment by janicduplessis** [comment]
Was there an issue with the rebase? You can try `git rebase --abort`.

**Comment by janicduplessis** [comment]
If that doesn't do it you can try to create a new branch from master then `git cherry-pick` your commits from this branch and then force push it back here.

**Comment by kevinejohn** [comment]
I just did another rebase master and force push and it seems to be displaying properly now on github. Not sure why it didn't work the first time

@astuetz good suggestion. I'll update it

**Comment by kevinejohn** [comment]
@janicduplessis I just added tests to UIExplorer and its fully working

https://github.com/kevinejohn/react-native/commit/aa8ad2f58011f0fc88b7ed90bb5ebdecd6e3a945

File: Examples/UIExplorer/ViewPagerAndroidExample.android.js [hunk 1]
```diff
@@ -166,6 +168,9 @@ var ViewPagerAndroidExample = React.createClass({
           {pages}
         </ViewPagerAndroid>
         <View style={styles.buttons}>
+          <Button enabled={true} text={this.state.swipeEnabled ? 'Swipe Enabled': 'Swipe Disabled'} onPress={() => this.setState({swipeEnabled: !this.state.swipeEnabled})}/>
```
Comment (janicduplessis) [inline Examples/UIExplorer/ViewPagerAndroidExample.android.js]: Nit: Put the props on separate lines here so the line isn't too long and add a space before the `:` in the ternary expression.

**Comment by kevinejohn** [comment]
@janicduplessis Fixed.

**Comment by mkonicek** [comment]
Does https://github.com/facebook/react-native/pull/5534 (merged) solve drawer locking?

**Comment by cubbuk** [comment]
Any progress on this PR and will this be merged anytime soon?
Thanks

**Comment by mkonicek** [comment]
@kevinejohn I believe #5534 might have already solved this? Do you still plan to work on this PR?

**Comment by kevinejohn** [comment]
@mkonicek Isn't DrawerLayoutAndroid is a completely different component then ViewPagerAndroid? If so this PR is still relevant

**Comment by astuetz** [comment]
@mkonicek: @kevinejohn is right, this PR is about locking a `ViewPager`, not a `DrawerLayout`

**Comment by mkonicek** [comment]
@kevinejohn Whoops! :) I read the title too quickly.

**Comment by mkonicek** [comment]
I see you're overriding both `onTouchEvent` and `onInterceptTouchEvent`?

For your test plan:

Can you add an example where you put a `TouchableHighlight` as well as a `ScrollView` inside the ViewPager with swipe disabled and show that both work?

**Comment by mkonicek** [comment]
> Whats the best way to test this directly from the repo?

This is the guide, feel free to improve it if it's not accurate / clear enough: https://facebook.github.io/react-native/docs/android-building-from-source.html#content

**Comment by shidhincr** [comment]
@kevinejohn Could you please update the PR ? It's nice to have this change to be merged.

**Comment by cnjon** [comment]
+1

**Comment by kevinejohn** [comment]
@shidhincr I just rebased.

**Comment by facebook-github-bot-1** [comment]
@mkonicek would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
@kmagiera Does this look good to you?

**Comment by skv-headless** [comment]
I think many people waiting for this.
@kevinejohn as I understood you need to make squash
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/
Does anything else block this PR?

**Comment by janicduplessis** [comment]
No need to squash the commit, it will be done by the shipit bot.

Code looks good to me, we just need to test the case @mkonicek mentioned to make sure touches are still handled properly.
> Can you add an example where you put a TouchableHighlight as well as a ScrollView inside the ViewPager with swipe disabled and show (using a video or a gif) that both work?

**Comment by kevinejohn** [comment]
@mkonicek  Here is the gif you asked for. Is there anything we need to do before merging this?

Test

**Comment by facebook-github-bot-2** [comment]
@mkonicek would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
Thanks for adding the gif @kevinejohn!

@facebook-github-bot shipit

**Comment by facebook-github-bot-2** [comment]
@kevinejohn updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@kevinejohn updated the pull request.

**Comment by facebook-github-bot-6** [comment]
@mkonicek would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by facebook-github-bot-0** [comment]
@mkonicek would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-7** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/493716290811666/int_phab) to review.

**Comment by ncnlinh** [comment]
Will this be in the next release yet?

**Comment by mkonicek** [comment]
@kevinejohn The code doesn't compile, see https://circleci.com/gh/facebook/react-native/5864

screen shot 2016-05-06 at 4 14 30 pm

**Comment by facebook-github-bot-2** [comment]
@kevinejohn updated the pull request.

**Comment by kevinejohn** [comment]
It looks like some of my work was duplicated in this commit:

https://github.com/facebook/react-native/commit/5a93877673105ca13c9e587796432881e26a0de7

I changed scrollEnabled to swipeEnabled. That commit doesn't fully connect it to a prop though so my PR is still relevant.

**Comment by facebook-github-bot-8** [comment]
@angelahess would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by angelahess** [comment]
scrollEnabled is actually connected through a prop already -- ViewPagerAndroid passes {...this.props} to <NativeAndroidViewPager />, which passes the scrollEnabled prop along. 

The functionality for scrollEnabled is already there, so this diff would just change the name and add an example.

**Comment by satya164** [comment]
I think `scrollEnabled` is better since there are other props like `onPageScroll` and `onPageScrollStateChanged`. Can you rebase to just add the example?

**Comment by facebook-github-bot-1** [comment]
@kevinejohn updated the pull request.

**Comment by kevinejohn** [comment]
@satya164 Just renamed it from "swipe" to "scroll" 
https://github.com/facebook/react-native/pull/5968/commits/1b6a51a81303d1698f86fce6f57f23b8b11b51b5

@angelahess You're right. This PR will just add the example now.

**Comment by facebook-github-bot-7** [comment]
@angelahess would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

File: ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java [hunk 2]
```diff
@@ -154,6 +158,7 @@ public boolean onTouchEvent(MotionEvent ev) {
     if (!mScrollEnabled) {
       return false;
     }
```
Comment (angelahess) [inline ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java]: If you are going to add a check to mScrollEnabled at the start of the method, delete this check here

File: ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPagerManager.java [hunk 3]
```diff
@@ -120,4 +114,9 @@ public void removeViewAt(ReactViewPager parent, int index) {
   public void setPageMargin(ReactViewPager pager, float margin) {
     pager.setPageMargin((int) PixelUtil.toPixelFromDIP(margin));
   }
+
+  @ReactProp(name = "scrollEnabled", defaultBoolean = true)
+  public void setScrollEnabled(ReactViewPager parent, boolean scrollEnabled) {
+    parent.setScrollEnabled(scrollEnabled);
+  }
```
Comment (angelahess) [inline ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPagerManager.java]: Revert this? Just to preserve blame.

File: Examples/UIExplorer/ViewPagerAndroidExample.android.js [hunk 4]
```diff
@@ -175,6 +177,12 @@ var ViewPagerAndroidExample = React.createClass({
           {pages}
         </ViewPagerAndroid>
         <View style={styles.buttons}>
+          <Button
+            enabled={true}
+            text={this.state.scrollEnabled ? 'Scroll Enabled' : 'Scroll Disabled'}
+            onPress={() => this.setState({scrollEnabled: !this.state.scrollEnabled})}/>
```
Comment (angelahess) [inline Examples/UIExplorer/ViewPagerAndroidExample.android.js]: /> on next line

File: ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java (hunk 2 above)
Comment (kevinejohn) [inline ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java]: That check is needed for both methods:
[onInterceptTouchEvent](https://github.com/kevinejohn/react-native/blob/1b6a51a81303d1698f86fce6f57f23b8b11b51b5/ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java#L144)
[onTouchEvent](https://github.com/kevinejohn/react-native/blob/1b6a51a81303d1698f86fce6f57f23b8b11b51b5/ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPager.java#L157)

File: ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPagerManager.java (hunk 3 above)
Comment (kevinejohn) [inline ReactAndroid/src/main/java/com/facebook/react/views/viewpager/ReactViewPagerManager.java]: Done

**Comment by facebook-github-bot-8** [comment]
@kevinejohn updated the pull request.

File: Examples/UIExplorer/ViewPagerAndroidExample.android.js (hunk 4 above)
Comment (kevinejohn) [inline Examples/UIExplorer/ViewPagerAndroidExample.android.js]: done
https://github.com/facebook/react-native/pull/5968/commits/446ed170fb8a70ff89379d8882e11577dcb03317

**Comment by angelahess** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-8** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/493716290811666/int_phab) to review.

**Comment by kinging123** [comment]
Hey, any updates about this? Was this ever merged?

**Pull request closed without merging**